.github/workflows/ci.yml

```
@@ -0,0 +1,166 @@
+name: CI
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  merge_group:
+
+jobs:
+  backward_compat:
+    name: "Backward Compatibility"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./pytest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          cache: pip
+      - uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43
+        with:
+          prefix-key: "0" # change this to invalidate CI cache
+          save-if: "false" # use the cache from nextest, but don’t double-save
```

The only remaining concern is  that the python script does not use the `--profile quick-release` (see `pytest/lib/branches.py:_compile_current`).

---

Fine if this is not resolved here, but a fixme would be good.
```suggestion
          save-if: "false" # use the cache from nextest, but don’t double-save
          # FIXME: these python scripts use a different cargo profile for the build.
```